docs/typescript/reference/dbos.md

```diff
@@ -0,0 +1,953 @@
+---
+sidebar_position: 4
+title: DBOS Class
+description: API reference for DBOS
+---
+
+# DBOS Class
+
+The `DBOS` class is the heart of the DBOS Transact programming API, providing decorators to identify key application functions, and accessors for context-dependent state.
+
+Note that this is the second major version of the API.  The first version of the API was based on passing [contexts](./contexts.md) and used a separate set of [decorators](./decorators.md).  It is fine to use both approaches in the same DBOS application, however the `DBOS` class is simpler and is therefore recommended.
+
+## Decorating Workflows, Transactions, and Steps
+DBOS provides reliability guarantees for programs that are [written as workflows comprised of steps, transactions, and child workflows](../programming-guide.md).  The workflow, transaction, and step functions are marked with decorators.
+
+### DBOS.workflow
+`@DBOS.workflow` registers a function as a DBOS workflow.  
+
+```typescript
+@DBOS.workflow(config?: WorkflowConfig)
+```
+
+When called, `workflow` functions are wrapped such that they will be [run to completion exactly once](../tutorials/idempotency-tutorial.md).
+
+#### Decorating Static Methods
+Workflow functions must be class members, and may be `static`.  To mark a `static` member function as a workflow, simply place the `@DBOS.workflow()` decorator above it.
+
+Example:
+```typescript
+import { DBOS, ConfiguredInstance, WorkflowQueue } from '@dbos-inc/dbos-sdk';
```

Comment:
```suggestion
import { DBOS } from '@dbos-inc/dbos-sdk';
```
I'd remove unused imports. Otherwise, it could confuse users.

Reply:
Yes, I reflowed this a few times, that import was supposed to be with the example that covers all sections below.  Fixing.